tests/meta/backup/conftest.py

```diff
@@ -1,7 +1,27 @@
+import fnmatch
+import subprocess
 from pathlib import Path
+from typing import Callable
 
 import pytest
 
+from cg.constants.pdc import PDCExitCodes
+
+
+@pytest.fixture
+def mock_pdc_query_method(archived_flow_cells) -> Callable:
+    """Returns a mock method mimicking the pattern search made by the dsmc q archive command."""
+
+    def mock_method(search_pattern: str) -> list[str]:
+        match = fnmatch.filter(archived_flow_cells, search_pattern)
+        if not match:
+            raise subprocess.CalledProcessError(
+                cmd="dummy_method", returncode=PDCExitCodes.NO_FILES_FOUND
+            )
+        return match
+
+    return mock_method
```

Comment:
I went a bit wild here. Let me know if it is too unclear 😅

Reply:
👍